aggregator/src/test/java/org/jenkinsci/plugins/workflow/CpsFlowExecutionTest.java

```diff
@@ -189,5 +190,30 @@ private static void assertStepExecutions(FlowExecution e, String... steps) throw
         }
         return r;
     }
+    @Test public void pause() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(
+                        "echo 'before'; semaphore 'one';  echo 'after';"));
+                WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+                SemaphoreStep.waitForStart("one/1", b);
+                CpsFlowExecution e = (CpsFlowExecution) b.getExecution();
+                e.pause(true);
+                e.waitForSuspension();
+                story.j.assertLogContains("before", b);
```

Comment:
`waitForSuspension` is basically deprecated. Use `JenkinsRule.waitForMessage` instead.